File: app/src/main/java/com/fdi/pad/pethouse/registration_user/activity_registration_password.java
package com.fdi.pad.pethouse.registration_user;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.fdi.pad.pethouse.R;

/**
 * Actividad que define el paso del registro donde se introduce la contraseña del usuario.
 */
public class activity_registration_password extends AppCompatActivity implements View.OnClickListener {
    /*------------------------------ATRIBUTOS----------------------------*/
    /**
     * Botón para ir al siguiente paso del registro del usuario.
     */
    private Button button_next;
    /**
     * Casillero de texto donde se encuentra el email del usuario.
     */
    private EditText edit_text_password;
    /**
     * Nombre del usuario.
     */
    private String name;
    /**
     * Apellidos del usuario.
     */
    private String surname;
    /**
     * Correo electrónico del usuario.
     */
    private String email;

    /*--------------------------ETAPAS---------------------------------*/
    /**
     * Creación de la actividad.
     *
     * @param savedInstanceState El estado de la aplicación en un paquete.
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_registration_password);

        button_next = (Button) findViewById(R.id.buttonNextRegistrationPassword);
        button_next.setOnClickListener(this);

        edit_text_password = (EditText) findViewById(R.id.editTextPasswordRegistrationPassword);

        /*Recibimos los datos del intent.*/
        Bundle b = getIntent().getExtras();

        name = (String) b.getString("name");
        surname = (String) b.getString("surname");
        email = (String) b.getString("email");
    }

    /*--------------------------EVENTOS---------------------------------*/
    /**
     * Ejecuta un procedimiento al tocar un elemento de la actividad.
     *
     * @param view Elemento que ha sido tocado.
     */
    @Override
    public void onClick(View view) {
        String password = edit_text_password.getText().toString();
        if (!validateForm(password)) {
            return;
        }
        Intent intent = new Intent(activity_registration_password.this, activity_registration_birthdate.class);
        intent.putExtra("name", name);
        intent.putExtra("surname", surname);
        intent.putExtra("email", email);
        intent.putExtra("password", password);
        startActivity(intent);
    }

    /*--------------------------MÉTODOS PRIVADOS---------------------------------*/
    /**
     * Comprueba que los casilleros no estén vacíos.
     *
     * @return Casilleros correctos
     */
    private boolean validateForm(String password) {
        boolean correct = true;

        if (TextUtils.isEmpty(password)) {
            edit_text_password.setError("Requerido.");
            correct = false;
        } else {
            edit_text_password.setError(null);
        }
        return correct;
    }
}

File: app/src/main/java/com/fdi/pad/pethouse/userRegistration/ActivityRegistrationPassword.kt
package com.fdi.pad.pethouse.userRegistration

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.text.TextUtils
import com.fdi.pad.pethouse.R
import kotlinx.android.synthetic.main.activity_registration_password.*
import java.util.regex.Pattern

/**
 * Actividad que define el paso del registro donde se introduce la contraseña del usuario.
 */
class ActivityRegistrationPassword : AppCompatActivity() {
    /*------------------------------CONSTANTES---------------------------*/
    /**
     * Parámetro para determinar el nombre del usuario.
     */
    private val nameUser = "name"
    /**
     * Parámetro para determinar los apellidos del usuario.
     */
    private val surnameUser  = "surname"
    /**
     * Parámetro para determinar el correo electrónico del usuario.
     */
    private val emailUser = "email"
    /**
     * Parámetro para determinar la contraseña del usuario.
     */
    private val passwordUser = "password"
    /**
     * Determina el patrón correcto de la contraseña del usuario.
     */
    private val patternPassword = ".{8,}"

    /*------------------------------ATRIBUTOS----------------------------*/
    /**
     * Nombre del usuario.
     */
    private var name: String? = null
    /**
     * Apellidos del usuario.
     */
    private var surname: String? = null
    /**
     * Correo electrónico del usuario.
     */
    private var email: String? = null

    /*--------------------------ETAPAS---------------------------------*/
    /**
     * Creación de la actividad.
     *
     * @param savedInstanceState El estado de la aplicación en un paquete.
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_registration_password)

        buttonNext.setOnClickListener { buttonNext() }

        /*Recibimos los datos del intent.*/
        name = intent.getStringExtra(nameUser)
        surname = intent.getStringExtra(surnameUser)
        email = intent.getStringExtra(emailUser)
    }

    /*--------------------------MÉTODOS PRIVADOS---------------------------------*/
    /**
     * Valida el campo y envía a la siguiente pantalla.
     */
    private fun buttonNext() {
        val password = editTextPassword.text.toString()
        if (!validateForm(password)) return

        val intent = Intent(this@ActivityRegistrationPassword, ActivityRegistrationBirthDate::class.java)
        intent.putExtra(nameUser, name)
        intent.putExtra(surnameUser, surname)
        intent.putExtra(emailUser, email)
        intent.putExtra(passwordUser, password)
        startActivity(intent)
    }

    /**
     * Comprueba que el campo tenga el formato correcto.
     *
     * @param password Contraseña del usuario.
     * @return Contraseña correcta correcta
     */
    private fun validateForm(password: String): Boolean {
        var correctPassword = true

        when {
            TextUtils.isEmpty(password) -> {
                editTextPassword.error = getString(R.string.required_field)
                correctPassword = false
            }
            Pattern.matches(patternPassword, password) -> editTextPassword.error = null

            else -> {
                editTextPassword.error = getString(R.string.incorrect_password_format)
                correctPassword = false
            }
        }
        return correctPassword
    }
}